### PR TITLE
Migrate to next js

### PR DESCRIPTION
**Pull request by andrew-t-james**
## Description

Migrate to [NextJS](https://nextjs.org/) to align web application with course curriculum.

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by mrbernnz** [comment]
@andrew-t-james Coups tout fil in the description?

File: src/components/Forms/FormAlert/FormAlert.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
+import { FaExclamationCircle } from 'react-icons/fa'
 import PropTypes from 'prop-types'
 
 function FormAlert({ errorMessage }) {
```
Comment (gixxerblade) [inline src/components/Forms/FormAlert/FormAlert.js:4]: Can you add a propType optional here for errorMessage?

File: src/components/Layout.js [hunk 2]
```diff
@@ -15,7 +14,7 @@ const TemplateWrapper = ({ children }) => (
         <Footer />
       </main>
     </ThemeProvider>
-  </>
+  </div>
 )
 
 TemplateWrapper.propTypes = {
```
Comment (gixxerblade) [inline src/components/Layout.js:20]: Probably: 
```js
children: PropTypes.element,
```

File: src/components/Toggle/Toggle.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { useContext } from 'react'
 import { ThemeContext } from '../../store/ThemeProvider'
 import { FaSun, FaMoon } from 'react-icons/fa'
 import PropTypes from 'prop-types'
-import './toggle.css'
+// import './toggle.css'
```
Comment (gixxerblade) [inline src/components/Toggle/Toggle.js:5]: Remove comment

**Review by gixxerblade — COMMENTED** [review]
Partially complete. Still a long way to go. Left off on `src/pages/contact.js`

File: src/components/Board/BoardCards.js [hunk 4]
```diff
@@ -1,57 +1,53 @@
...
-      return 1
-    }
-    return 0
-  })
 
+function BoardCards({ boardMemberCollection }) {
   return (
     <>
-      {sortedBoardMembers.map(({ node }) => {
+      {boardMemberCollection.map(boardMember => {
```
[the remark this answers is not in the document]
Comment (andrew-t-james) [reply src/components/Board/BoardCards.js:8]: Sorting happens before the render see [here](https://github.com/Vets-Who-Code/vets-who-code-app/pull/325/files#diff-19b79f3fe4024aabfa6d7d716cb558fc9d3befd8042508001dc70fb3052a8cf5R34)

File: src/components/Forms/ApplyForm/ApplyForm.js [hunk 5]
```diff
@@ -1,88 +1,72 @@
...
-        setZipError('Invalid zipcode')
+        onSubmitError('OOPS! Something went wrong. Please try again later.')
       }
-    } catch (error) {
-      setLoading(false)
-      onSubmitError('OOPS! Something went wrong. Please try again later.')
     }
-  }
+    fetchCityState()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
[the remark this answers is not in the document]
Comment (andrew-t-james) [reply src/components/Forms/ApplyForm/ApplyForm.js:47]: The lint rule broke the render 🤷🏼

[the remark this answers is not in the document]
Comment (gixxerblade) [reply src/components/Forms/ApplyForm/ApplyForm.js:47]: Maybe add cityState, error, zipError to the list of deps?

[the remark this answers is not in the document]
Comment (andrew-t-james) [reply src/components/Forms/ApplyForm/ApplyForm.js:47]: When adding any deps it broke all the things.

**Review by jeromehardaway — APPROVED** [review]

**Pull request merged**